carma_cooperative_perception/include/carma_cooperative_perception/j2735_types.hpp

```diff
@@ -0,0 +1,145 @@
+#ifndef CARMA_COOPERATIVE_PERCEPTION_J2735_TYPES_HPP_
+#define CARMA_COOPERATIVE_PERCEPTION_J2735_TYPES_HPP_
+
+/*
+ * Copyright 2023 Leidos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <units.h>
+
+#include <j2735_v2x_msgs/msg/acceleration_set4_way.hpp>
+#include <j2735_v2x_msgs/msg/d_date_time.hpp>
+#include <j2735_v2x_msgs/msg/heading.hpp>
+#include <j2735_v2x_msgs/msg/position3_d.hpp>
+#include <j2735_v2x_msgs/msg/speed.hpp>
+#include <optional>
+
+#include "carma_cooperative_perception/month.hpp"
+#include "carma_cooperative_perception/units_extensions.hpp"
+
+namespace carma_cooperative_perception
+{
+
+struct DDateTime
```

Comment:
I am conflicted with header-only libraries for non-template code. The traditional approach, like you and online posts mention, is to put definitions in `.cpp` files. However, head-only libraries seem to be in vogue. My thought process it:

1. All files except `sdsm_to_detection_list_node.*` should be moved to a separate library in the future because they have usefulness outside of this package. Keeping them header-only makes the migration easier. All the code is in one place. Once it is moved, we can revisit the header/source split.
2. As you mentioned, the compile times are the biggest issue. However, this package does not get linked to anything else. I'm skeptical of the compile time argument here, but I can time the compile times with the more traditional implementation.